src/main/java/gyro/google/compute/InstanceResource.java

```diff
@@ -0,0 +1,316 @@
+     */
+    @Regex("[a-z]([-a-z0-9]*[a-z0-9])?")
+    @Required
+    public String getName() {
+        return name;
+    }
+
+    public void setName(String name) {
+        this.name = name;
+    }
+
+    /**
+     * Name of zone the where the instance resides, i.e., ``us-central1-a``. See `available region and zones<https://cloud.google.com/compute/docs/regions-zones/#available/>`_. for the current list of zones.
+     */
+    @Required
+    public String getZone() {
+        return zone;
+    }
+
+    public void setZone(String zone) {
+        this.zone = zone;
+    }
+
+    /**
+     * An optional description for the resource.
+     */
+    @Output
+    public String getDescription() {
+        return description;
+    }
+
+    public void setDescription(String description) {
+        this.description = description;
+    }
+
+    /**
+     * Full or partial URL of the machine type resource to use for this instance, in the format: zones/zone/machineTypes/machine-type. See `creating custom machine types <https://cloud.google.com/compute/docs/instances/creating-instance-with-custom-machine-type#specifications/>`_.
+     */
+    public String getMachineType() {
+        return machineType;
+    }
+
+    public void setMachineType(String machineType) {
+        this.machineType = machineType != null ? machineType.substring(machineType.lastIndexOf("zones/")) : null;
+    }
+
+    /**
+     * List of network configurations for this instance. These specify how interfaces are configured to interact with other network services, such as connecting to the internet. Multiple interfaces are supported.
+     */
+    public List<InstanceNetworkInterface> getNetworkInterfaces() {
+        if (networkInterfaces == null) {
+            networkInterfaces = new ArrayList<>();
+        }
+        return networkInterfaces;
+    }
+
+    public void setNetworkInterfaces(List<InstanceNetworkInterface> networkInterfaces) {
+        this.networkInterfaces = networkInterfaces;
+    }
+
+    /**
+     * List of disks associated with this instance. Persistent disks must be created before you can assign them.
+     */
+    @Output
+    public List<InstanceAttachedDisk> getDisks() {
+        if (disks == null) {
+            disks = new ArrayList<>();
+        }
+        return disks;
+    }
+
+    public void setDisks(List<InstanceAttachedDisk> disks) {
+        this.disks = disks;
+    }
+
+    /**
+     * URL to the instance.
+     */
+    @Id
+    @Output
+    public String getSelfLink() {
+        return selfLink;
+    }
+
+    public void setSelfLink(String selfLink) {
+        this.selfLink = selfLink;
+    }
+
+    /**
+     * Map of key-value pairs to apply to this instance. May be updated after creation.
```

Comment:
We dont generally say `maybe updated` on any updatable fields.
```suggestion
     * Map of key-value pairs to apply to this instance.
```

Reply:
Updated